front-end/src/views/user/Login.vue

```diff
@@ -0,0 +1,232 @@
+      }
+    }
+  },
+  created () {
+    const needRememberMe = storage.get(REMEMBER_ME)
+    if (needRememberMe === 'true') {
+      this.needRememberMe = true
+    }
+  },
+  methods: {
+    ...mapActions(['Login', 'Logout']),
+    handleUsernameOrEmail (rule, value, callback) {
+      const { state } = this
+      const regex = /^([a-zA-Z0-9_-])+@([a-zA-Z0-9_-])+((\.[a-zA-Z0-9_-]{2,3}){1,2})$/
+      if (regex.test(value)) {
+        state.loginType = 0
+      } else {
+        state.loginType = 1
+      }
+      callback()
+    },
+    handleSubmit (e) {
+      e.preventDefault()
+      const { form: { validateFields }, state, Login } = this
+
+      state.loginBtn = true
+
+      validateFields(['username', 'password'], { force: true }, (err, values) => {
+        if (!err) {
+          const loginParams = { ...values, password: md5(values.password) }
+
+          if (this.needRememberMe) {
+            storage.set(REMEMBER_ME, 'true')
+            storage.set(LOGIN_INFO, JSON.stringify(loginParams))
+          } else {
+            storage.set(REMEMBER_ME, 'false')
+          }
+
+          Login(loginParams)
+            .then(res => this.loginSuccess(res))
+            .catch(err => this.requestFailed(err))
+            .finally(() => {
+              state.loginBtn = false
+            })
+        } else {
+          setTimeout(() => {
+            state.loginBtn = false
+          }, 600)
+        }
+      })
+    },
+    getCaptcha (e) {
+      e.preventDefault()
+      const { form: { validateFields }, state } = this
+
+      validateFields(['mobile'], { force: true }, (err, values) => {
+        if (!err) {
+          state.smsSendBtn = true
+
+          const interval = window.setInterval(() => {
+            if (state.time-- <= 0) {
+              state.time = 60
+              state.smsSendBtn = false
+              window.clearInterval(interval)
+            }
+          }, 1000)
+
+          const hide = this.$message.loading('验证码发送中..', 0)
+          getSmsCaptcha({ mobile: values.mobile }).then(res => {
+            setTimeout(hide, 2500)
+            this.$notification['success']({
+              message: '提示',
+              description: '验证码获取成功，您的验证码为：' + res.result.captcha,
+              duration: 8
+            })
+          }).catch(err => {
+            setTimeout(hide, 1)
+            clearInterval(interval)
+            state.time = 60
+            state.smsSendBtn = false
+            this.requestFailed(err)
+          })
+        }
+      })
+    },
+    loginSuccess () {
+      this.$router.push({ path: '/' })
+      // Delayed by 1s to show the welcome pop
+      setTimeout(() => {
+        this.$notification.success({
+          message: '欢迎',
+          description: `${ timeFix() }，欢迎回来`
+        })
+      }, 1000)
+      this.isLoginError = false
+    },
+    requestFailed (err) {
+      this.isLoginError = true
+      this.$notification['error']({
```

Comment:
Codacy has a fix for the issue: [["error"] is better written in dot notation.](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834763)
```suggestion
      this.$notification.error({
```